### DISCUSSION TITLE
Is it possible to keep the capitalization of HTML nodes?

### DISCUSSION THREAD

**Discussion by Kesocial** [programble/scraper#179] [Q&A] [answered]
Hello!!!
I need the labels to remain capitalized, is this possible?
I read something about `scraper::CaseSensitivity` in the docs, but it's not very clear to me, thanks in advance!

### Example:
**Input html:**
```
<NodeMain> 
  <OtherNode>   
  </OtherNode> 
</NodeMain>
```

**Output:**_(Values of elements)_
```
<nodemain>
<othernode> 
``` 

**Output keeping the capitalization:**_(Values of elements)_
```
<NodeMain>
<OtherNode> 
```

**Comment by cfvescovo** [comment, marked as answer]
`scraper::CaseSensitivity` is used by `Element::has_id` and `Element::has_class` when checking if an element contains a specific id or class so it's not related to what you get as element values. That depends on HTML serialization and, as element tag names are case insensitive (see https://w3c.github.io/html-reference/documents.html), I think you should avoid treating them as case sensitive

**Comment by drauschenbach** [comment]
This is a deal-breaker for me, as I'm trying to manipulate Leptos and Ember style fragments that are component references such as `<MyComponent arg1="" arg2="" />`. For dom manipulation, I was going to do this:

```rust
// Parse
let mut document = Html::parse_fragment(&body);
let tree = HtmlTreeSink::new(document);

// Mutate
// TODO perform transformations

// Finalize
let document = tree.finish();
let body = document.html();
```

But this library is not safe for dom manipulation if it's going to perform its own arbitrary transformations.
